src/db.c

```diff
@@ -2184,6 +2184,64 @@ int sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *
     return result->numkeys;
 }
 
+/* Helper function to extract keys from the SORT command.
+ * Unlike sortGetKeys this command will also get the access key patterns
+ * provided via 'get' and 'by' options. the '#' used by 'get' is not referring to any key
+ * so it is skipped.
+ * This is implemented only for the use of ACL authorization verification.
+ *
+ * SORT <sort-key> ... by <sort-by-key-pattern> ... get <get-key-patter> ...
+ *
+ * The first argument of SORT is always a key, however a list of options
+ * follow in SQL-alike style.
+ *
+ * This command declares incomplete keys, so the flags are correctly set for this function */
+int sortGetKeysWithPatterns(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *result) {
```

Comment:
Can we refactor most of this code into a helper method? then have sortGetKeysWith/WithoutPatterns

Reply:
fixed in [3852f8e](https://github.com/redis/redis/pull/10340/commits/3852f8e03e033b914775a7b68a62ca554041f9b3)